pkg/cache/task.go

```diff
@@ -42,14 +43,15 @@ type Task struct {
 	pod            *v1.Pod
 	kubeClient     client.KubeClient
 	schedulerApi   api.SchedulerApi
+	volumeBinder   *volumebinder.VolumeBinder
 	nodeName       string
 	sm             *fsm.FSM
 	lock           *sync.RWMutex
 }
 
-func newTask(tid string, app *Application, client client.KubeClient, schedulerApi api.SchedulerApi, pod *v1.Pod) Task {
+func newTask(tid string, app *Application, client client.KubeClient, schedulerApi api.SchedulerApi, volumeBinder *volumebinder.VolumeBinder, pod *v1.Pod) Task {
```

Comment:
Instead of passing in three different references it would be much easier to pass in the `context` and do the calls using the `context` values. Cleans up the signatures of a number of calls.

Reply:
Fair point, changed to pass in context instead in latest commit.